smarts/core/waypoints.py

```diff
@@ -76,6 +74,30 @@ def signed_lateral_error(self, p):
     def pose(self):
         return Pose.from_center(tuple(self.pos), self.heading)
 
+    def __hash__(self):
```

Comment:
Is the built-in hash and equal from `dataclass(...)` not sufficient?

Reply:
Ideally it is so we don't need to maintain our own.

Reply:
Thanks, If we only implement hash then for some of the tests assertions, we need to use expressions like `hash(some_wp)==hash(other_wp)` as it is mentioned by @davidrusu , by defining `eq` then those expression simplifies to `some_wp==other_wp`.

Reply:
Thanks, unfortunately, the use of numpy array in the waypoints results in ambiguity error for the built in hash, that's why I had to implement our version of the hash.

Reply:
> Looks good, excited about getting this in :)

Thank you so much, @JVillella